bql/semantic/expression.go

```diff
@@ -81,37 +81,44 @@ func (o OP) String() string {
 	}
 }
 
-// formatCell formats a given cell into a trimmed comparable string
-func formatCell(c *table.Cell) string {
+// formatCell formats a given cell into a trimmed comparable string.
+func formatCell(c *table.Cell) (string, error) {
 	if c.L != nil {
-		return strings.TrimSpace(c.L.ToComparableString())
+		return strings.TrimSpace(c.L.ToComparableString()), nil
 	}
-	return strings.TrimSpace(c.String())
+	if c.S != nil {
+		formatted, err := literal.DefaultBuilder().Build(literal.Text, *c.S)
+		if err != nil {
+			return "", fmt.Errorf("could not build a literal from string while trying to formatCell, got error: %v", err)
+		}
+		return strings.TrimSpace(formatted.ToComparableString()), nil
+	}
+	return strings.TrimSpace(c.String()), nil
 }
 
 // evaluationNode represents the internal representation of one expression.
 type evaluationNode struct {
-	op OP // operation
+	op OP // operation.
 
-	lB string // left binding
-	rB string // right binding
+	lB string // left binding.
+	rB string // right binding.
 }
 
 // Evaluate the expression.
 func (e *evaluationNode) Evaluate(r table.Row) (bool, error) {
-	// Binary evaluation
+	// Binary evaluation.
 	eval := func() (*table.Cell, *table.Cell, error) {
 		var (
 			eL, eR *table.Cell
 			ok     bool
 		)
 		eL, ok = r[e.lB]
 		if !ok {
-			return nil, nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.lB, r)
+			return nil, nil, fmt.Errorf("comparison operation requires the binding value for %q for row %q to exist", e.lB, r)
```

Comment:
`%q` should be used for strings, but `r` is a table.Row. Either use `%v` or convert it to string before (here and below).

Reply:
Very well pointed, I have just fixed it. Thank you Tati.